core/src/main/scala/org/apache/spark/deploy/master/ui/ApplicationPage.scala

```diff
@@ -43,8 +43,9 @@ private[ui] class ApplicationPage(parent: MasterWebUI) extends WebUIPage("app")
       return UIUtils.basicSparkPage(msg, "Not Found")
     }
 
-    val executorHeaders = Seq("ExecutorID", "Worker", "Cores", "Memory", "State", "Logs")
+    val executorHeaders = Seq("Executor ID", "Worker", "Cores", "Memory", "State", "Logs")
     val allExecutors = (app.executors.values ++ app.removedExecutors).toSet.toSeq
+      .sortWith(_.id < _.id)
```

Comment:
@srowen actually Scala's `List` is a linked list.  You may want `toIndexedSeq`.  In any case, I think all of them will dump the data into an array for sorting (even `ArrayBuffer`), so it probably doesn't matter.